tests/framework/cli/pipeline/test_pipeline.py

```diff
@@ -603,3 +603,16 @@ def test_sync_no_target(self, source, tmp_path):
         assert (target / "existing" / "common").read_text(encoding="utf-8") == "source"
         assert not (target / "existing" / "target_file").exists()
         assert (target / "new" / "source_file").is_file()
+
+    def test_sync_file_source(self, tmp_path):
```

Comment:
You are probably wondering _why_ a PR that removes a feature is also introducing a new unit test for an unrelated feature. 

This is because, for some reason, there was one [specific conditional](https://github.com/kedro-org/kedro/blob/c8c60aa01d640b9174181f016a975ddf39da06cb/kedro/framework/cli/pipeline.py#L277) in the `_sync_dirs()` function in the pipeline CLI that's only being covered by the micropackaging tests.